src/contract_wrappers/tokens/security_token_wrapper.ts

```diff
@@ -994,6 +999,13 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     await this.checkOnlyOwner(params.txData);
     await this.checkModuleCostBelowMaxCost(params.address, maxCost);
     await this.checkModuleStructAddressIsEmpty(params.address);
+    if(params.budget) {
```

Comment:
It is not ok. 
You should get module cost from module factory and compare its value with the balance. 
I guess we are getting this cost on `this.checkModuleCostBelowMaxCost` so maybe it's easier to add this requirement there.

Reply:
In this case i am going to forget about budget and make a change inside `checkModuleCostBelowMaxCost`